src/transformStateWithClones.js

```diff
@@ -7,7 +7,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let currentState = { ...state };
+  const result = [];
+
+  for (const action of actions) {
+    let newState = { ...currentState };
+
+    if (action.type === 'addProperties') {
+      for (const key in action.extraData) {
+        newState[key] = action.extraData[key];
+      }
+    }
+
+    if (action.type === 'removeProperties') {
+      for (const key of action.keysToRemove) {
+        delete newState[key];
+      }
+    }
+
+    if (action.type === 'clear') {
+      newState = {};
+    }
+    
+    currentState = newState;
+    result.push(newState);
```

Comment:
Instead of using multiple if statements, consider using a switch statement since you have a limited number of conditions (action types) to check. This will make your code cleaner and more maintainable.